tools/cli/lib/parseargs.c

```diff
@@ -335,64 +322,31 @@ ParseOption(
             pszToken = strtok_r(NULL, ",:", &pszTokenSave);
         }
     }
-    else if(!strcasecmp(pszName, "reboot-required"))
-    {
-        dwError = AddSetOptWithValues(
-                      pCmdArgs,
-                      CMDOPT_KEYVALUE,
-                      pszName,
-                      "1");
-        BAIL_ON_CLI_ERROR(dwError);
-    }
     else if(!strcasecmp(pszName, "installroot"))
     {
         dwError = TDNFAllocateString(
                       optarg,
                       &pCmdArgs->pszInstallRoot);
-        BAIL_ON_CLI_ERROR(dwError);
     }
     else if(!strcasecmp(pszName, "releasever"))
     {
         dwError = TDNFAllocateString(
                       optarg,
                       &pCmdArgs->pszReleaseVer);
-        BAIL_ON_CLI_ERROR(dwError);
     }
-    else if(!strcasecmp(pszName, "skipconflicts"))
+    else if(!strcasecmp(pszName, "skipconflicts") ||
+            !strcasecmp(pszName, "skipobsoletes") ||
+            !strcasecmp(pszName, "skipsignature") ||
+            !strcasecmp(pszName, "skipdigest") ||
+            !strcasecmp(pszName, "noplugins") ||
+            !strcasecmp(pszName, "security") ||
+            !strcasecmp(pszName, "reboot-required"))
```

Comment:
would it be better if we make a local struct with args that needs to be passed to AddSetOptWithValues 
like 
struct setopt_args
{
TDNF_CMDOPT_TYPE type;
char *optarg;
char *optval;
} optvaltable[] =
{
{CMDOPT_KEYVALUE, "sec-severity",  optarg},
{CMDOPT_KEYVALUE, "security",  "1"},
};

etc and rewrite this function to loop through the table and just handle the extra validation of comma separated strings (in exclude for eg) as needed?

Reply:
Good idea but how about: https://github.com/vmware/tdnf/blob/ae364836529ec020d66ac1af86590c70f2711332/common/setopt.c#L352?

Reply:
It can be done, ignore my previous comment. Was thinking about some different solution of passing a struct as function parameter